sdk/python/tests/expediagroup/test_milvus_online_store.py

```diff
@@ -1,37 +1,346 @@
+        # Disconnecting from Milvus
+        connections.disconnect("milvus")
+
+
+class TestMilvusConnectionManager:
+    def test_connection_manager(self, repo_config, caplog, milvus_online_setup, mocker):
+
+        mocker.patch("pymilvus.connections.connect")
+        with MilvusConnectionManager(repo_config.online_store):
+            assert (
+                f"Connecting to Milvus with alias {repo_config.online_store.alias} and host {repo_config.online_store.host} and default port {repo_config.online_store.port}."
+                in caplog.text
+            )
+
+        connections.connect.assert_called_once_with(
+            host=repo_config.online_store.host,
+            username=repo_config.online_store.username,
+            password=repo_config.online_store.password,
+            use_secure=True,
+        )
+
+    def test_context_manager_exit(
+        self, repo_config, caplog, milvus_online_setup, mocker
+    ):
+        # Create a mock for connections.disconnect
+        mock_disconnect = mocker.patch("pymilvus.connections.disconnect")
+
+        # Create a mock logger to capture log calls
+        mock_logger = mocker.patch(
+            "feast.expediagroup.vectordb.milvus_online_store.logger", autospec=True
+        )
+
+        with MilvusConnectionManager(repo_config.online_store):
+            print("Doing something")
+
+        # Assert that connections.disconnect was called with the expected argument
+        mock_disconnect.assert_called_once_with(repo_config.online_store.alias)
+
+        with pytest.raises(Exception):
+            with MilvusConnectionManager(repo_config.online_store):
+                raise Exception("Test Exception")
+        mock_logger.error.assert_called_once()
+
+
+class TestMilvusOnlineStore:
+
+    collection_to_write = "Collection2"
+    collection_to_delete = "Collection1"
+
+    def setup_method(self, milvus_online_setup):
+        # Ensuring that the collections created are dropped before the tests are run
+        with PymilvusConnectionContext():
+            # Dropping collections if they exist
+            if utility.has_collection(self.collection_to_delete):
+                utility.drop_collection(self.collection_to_delete)
+            if utility.has_collection(self.collection_to_write):
+                utility.drop_collection(self.collection_to_write)
+            # Closing the temporary collection to do this
+
+    def test_milvus_update_add_collection(
+        self, repo_config, milvus_online_setup, caplog
+    ):
+
+        schema_vector = [
+            Field(
+                name="feature2",
+                dtype=Int64,
+                tags={"is_primary": "True", "description": "int64"},
+            ),
+            Field(
+                name="feature1",
+                dtype=Array(Float32),
+                tags={
+                    "is_primary": "False",
+                    "description": "float32",
+                    "dimension": "128",
```

Comment:
thinking about it this is actually a great suggestion to use a tag to pass along the number of dimensions. I think we can do likewise with index algorithm and eliminate the need for a `VectorFeatureView` altogether. What do you think?

Reply:
Yes, thats a good idea